### ISSUE TITLE
EACCES: permission denied on MacOS

### ISSUE THREAD

**Issue by martin-braun** [oxen-io/session-desktop#2772]
**Describe the bug**
On MacOS, `yarn install --frozen-lockfile` fails with: 

```
> yarn install --frozen-lockfile
yarn install v1.22.19
[1/6] 🔍  Validating package.json...
[2/6] 🔍  Resolving packages...
[3/6] 🚚  Fetching packages...
[###########################################----------------------------------------------------
[4/6] 🔗  Linking dependencies...
warning " > mic-recorder-to-mp3@2.2.2" has unmet peer dependency "webrtc-adapter@>=4.1.1".
warning " > react-virtualized@9.22.3" has incorrect peer dependency "react@^15.3.0 || ^16.0.0-alpha".
warning " > react-virtualized@9.22.3" has incorrect peer dependency "react-dom@^15.3.0 || ^16.0.0-alpha".
warning " > styled-components@5.1.1" has unmet peer dependency "react-is@>= 16.8.0".
warning " > node-loader@2.0.0" has unmet peer dependency "webpack@^5.0.0".
warning " > ts-loader@9.3.0" has unmet peer dependency "webpack@^5.0.0".
warning " > tslint-microsoft-contrib@6.0.0" has incorrect peer dependency "typescript@^2.1.0 || ^3.0.0".
[5/6] 🔨  Building fresh packages...
[1/11] ⠐ better-sqlite3
[2/11] ⠐ dtrace-provider
[-/11] ⠐ waiting...
[7/11] ⠐ electron
error /Users/marty/....../session-desktop/node_modules/electron: Command failed.
Exit code: 1
Command: node install.js
Arguments:
Directory: /Users/marty/....../session-desktop/node_modules/electron
Output:
Error: EACCES: permission denied, mkdir '/Users/
```

**To Reproduce**
Follow the steps in `CONTRIBUTING.md` on MacOS.

I think this is related to the fact that `/Users/` does not belong to `$(whoiam)`:

```
drwxr-xr-x     - root  2022-08-24 10:59  Users/
```

**Screenshots or Logs**

```
> neofetch                                                                                               [clearnet]
                    'c.          marty@Martins-MacBook-Pro.local
                 ,xNMM.          -------------------------------
               .OMMMMo           OS: macOS 12.6 21G115 x86_64
               OMMM0,            Host: MacBookPro16,1
     .;loddo:' loolloddol;.      Kernel: 21.6.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 1 day, 53 mins
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 322 (brew), 98 (nix-user)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8.1
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 2048x1280@2x
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Amethyst
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    Terminal: tmux
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   CPU: Intel i9-9980HK (16) @ 2.40GHz
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   GPU: Intel UHD Graphics 630, AMD Radeon Pro 5500M
    kMMMMMMMMMMMMMMMMMMMMMMd     Memory: 19294MiB / 65536MiB
     ;KMMMMMMMWXXWMMMMMMMk.
       .cooc,.    .,coo:.
```

**Other information (please complete the following information):**

* Device: see above
* OS: see above
* Session Version or Git commit hash: 537724cfdb3c701a3ea80cfdbbaaed0063f6e109

**Comment by yougotwill** [comment]
Hi @martin-braun how have you installed yarn on your machine and what terminal are you using?

My gut feeling is that this is a folder access issue.

**Comment by martin-braun** [comment]
@yougotwill Thanks for your quick response.

> My gut feeling is that this is a folder access issue.

I was very certain that it's a folder access issue, but the folder that was reported was `/Users` that does belong to `root` (and it has to). That's the thing that threw me off. 

But then it hit me, because the error message `Error: EACCES: permission denied, mkdir '/Users/` is cut off (missing `'`).

I did a recursive `chown` on the project folder and retried and it revealed that it failed to do `mkdir '/Users/...../Library/Caches/electron/a994acd32232923a`. Not sure why `tmux` swallowed most of the path previously.

I think I sometime ago I built something with Electron on sudo maybe, I purged that `electron` directory and retried, and luckily we have a success story now.

Thanks!
